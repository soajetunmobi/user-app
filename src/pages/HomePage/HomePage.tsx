import React, { FunctionComponent } from 'react'
import { useUserContext } from 'providers'
import { LoadingIndicator } from 'components/shared/LoadingIndicator'
import { useHistory } from 'react-router-dom'
import { routes } from 'const/routes'
import { Button } from '@material-ui/core'

export const HomePage: FunctionComponent = () => {
  const { user } = useUserContext()
  const { id, first_name } = user
  const history = useHistory()

  if (!user) {
    return <LoadingIndicator />
  }

  const login = () => {
    history.push(routes.login)
  }

  const renderContent = () => {
    if (id.length == 0) {
      return (
        <div className='text-align--center'>
          <h4 data-testid='login-welcome'>Welcome, please login to access your profile</h4>
          <Button color='primary' data-testid='login-button' onClick={login}>
            Login
          </Button>
        </div>
      )
    }
    return <h4 data-testid='logged-in-welcome'>Welcome {first_name}</h4>
  }

  return <div>{renderContent()}</div>
}
