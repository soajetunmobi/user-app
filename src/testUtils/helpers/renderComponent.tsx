import React, { ReactElement, ReactNode } from 'react'
import { render } from '@testing-library/react'
import { UserContext } from 'providers'

const renderComponent = (component: ReactElement, user: User): ReactNode => {
  return render(
    <UserContext.Provider value={{ user, setUser: user => user }}>{component}</UserContext.Provider>
  )
}

export default renderComponent
