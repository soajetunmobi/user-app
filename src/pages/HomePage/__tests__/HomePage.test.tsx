import React from 'react'
import renderer from 'react-test-renderer'
import { HomePage } from '../HomePage'
import { screen } from '@testing-library/react'
import renderComponent from 'testUtils/helpers/renderComponent'
import { MOCKED_USER } from 'testUtils/mocks/user'

describe('HomePage', () => {
  it('renders home page', () => {
    const root = renderer.create(<HomePage />).toJSON()
    expect(root).toMatchSnapshot()
  })

  describe('Dynamic content in home page', () => {
    it('should show login link if user is not logged in', () => {
      renderComponent(<HomePage />, { ...MOCKED_USER, id: '', email: '' })
      expect(screen.getByTestId('login-button')).toBeInTheDocument()
      expect(screen.getByTestId('login-welcome')).toBeInTheDocument()
    })

    it('should show welcome text when user is logged in', () => {
      renderComponent(<HomePage />, MOCKED_USER)
      expect(screen.getByTestId('logged-in-welcome')).toBeInTheDocument()
    })

    it('should salute user when logged in', () => {
      renderComponent(<HomePage />, MOCKED_USER)
      expect(screen.getByTestId('logged-in-welcome')).toBeInTheDocument()
      expect(screen.getByText(`Welcome ${MOCKED_USER.first_name}`)).toBeInTheDocument()
    })
  })
})
