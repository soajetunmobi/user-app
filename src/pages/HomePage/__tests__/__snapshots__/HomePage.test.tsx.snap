// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage renders home page 1`] = `
<div>
  <div
    className="text-align--center"
  >
    <h4
      data-testid="login-welcome"
    >
      Welcome, please login to access your profile
    </h4>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
      data-testid="login-button"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Login
      </span>
    </button>
  </div>
</div>
`;
